src/Mvc/Mvc.TagHelpers/src/DependencyInjection/TagHelperExtensions.cs

```diff
@@ -84,6 +84,7 @@ internal static void AddCacheTagHelperServices(IServiceCollection services)
 
         // Required default services for cache tag helpers
         services.AddDistributedMemoryCache();
+        services.AddMemoryCache(options => options.TrackLinkedCacheEntries = true);
```

Comment:
This shouldn't be part of MVC if it's just for the CacheTagHelper_BubblesExpirationOfNestedTagHelpers test site.

Reply:
This restores the behavior MVC cached tag helpers had prior to 7.0, it's not just for the test site. Without this, each site would need to opt in. @pranavkm?

Reply:
The test site is using a memory cache which is why I think you "need" this. https://github.com/dotnet/aspnetcore/blob/c47616629b5b3f742bc003ef63fbf480d00828ff/src/Mvc/test/WebSites/HtmlGenerationWebSite/ProductsService.cs#L29

Reply:
/cc @sebastienros since he's familiar with this tag helper. 

Some context, as part of https://github.com/dotnet/runtime/issues/45592, `MemoryCache` no longer adds expiration tokens added to a child entry to it's parent. Consider the following case:

```razor
<cache vary-by-header="User-Agent">
    Outer cache @DateTime.Now
    <cache expires-after="@TimeSpan.FromMinutes(1)">
        Last updated: @DateTime.Now
    </cache>
</cache>
````

In this case, even though the inner cache is meant to expire every minute, the view's output will be perpetually cached based on the user-agent. This seems like it'll be hard for us to explain to our users and given we currently don't let them configure the cache, it seems like more work to figure out how to allow setting this. Defaulting to the previous behavior seems fine given we haven't seen any massive complaints about this resulting in the tag helper being less compelling for our users.

FYI @adamsitnik

Reply:
AddDistributedMemoryCache was already adding MemoryCache.

Reply:
> What's being done here though is adding MemoryCache with TrackLinkedCacheEntries to true when adding MVC, outside of the CacheTagHelper context

This is part of the `AddCacheTagHelperServices` call no? Am I missing something?

Reply:
> This is part of the AddCacheTagHelperServices call no? Am I missing something?

@pranavkm, the cache tag helper uses a memory cache from `CacheTagHelperMemoryCacheFactory` hence why it's being updated as part of this PR.

> AddDistributedMemoryCache was already adding MemoryCache.

@Tratcher where? https://source.dot.net/#Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs,b69d3c9545a8a300

Reply:
Found where MemoryCache is being added
https://github.com/dotnet/aspnetcore/blob/0c840a17fd42bc00cd379eb8ce4006eb5ba6dab1/src/Mvc/Mvc.Razor/src/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs#L172

Reply:
Ah, my bad, MemoryDistributedCache creates it's own instance.
https://source.dot.net/#Microsoft.Extensions.Caching.Memory/MemoryDistributedCache.cs,33

Reply:
Hmm, I wonder what this cache is used for. Sorry I hadn't looked at the factory and assumed this configured the memory cache used by CacheTagHelper. Let's start with leaving this be and we'll see what falls out?